python/lsst/resources/_resourceHandles/_baseResourceHandle.py

```diff
@@ -0,0 +1,178 @@
+U = TypeVar("U", str, bytes)
+
+
+class CloseStatus(Enum):
+    """Enumerated closed/open status of a file handle, implementation detail
+    that may be used by BaseResourceHandle children.
+    """
+
+    OPEN = auto()
+    CLOSING = auto()
+    CLOSED = auto()
+
+
+class ResourceHandleProtocol(Protocol, Generic[U]):
+    """Defines the interface protocol that is compatible with children of
+    `BaseResourceHandle`.
+
+    Any class that satisfies this protocol can be used in any context where a
+    `BaseResourceHandle` is expected.
+    """
+
+    @abstractproperty
+    def mode(self) -> str:
+        ...
+
+    @abstractmethod
+    def close(self) -> None:
+        ...
+
+    @abstractproperty
+    def closed(self) -> bool:
+        ...
+
+    @abstractmethod
+    def fileno(self) -> int:
+        ...
+
+    @abstractmethod
+    def flush(self) -> None:
+        ...
+
+    @abstractproperty
+    def isatty(self) -> Union[bool, Callable[[], bool]]:
+        ...
+
+    @abstractmethod
+    def readable(self) -> bool:
+        ...
+
+    @abstractmethod
+    def readline(self, size: int = -1) -> U:
+        ...
+
+    @abstractmethod
+    def readlines(self, hint: int = -1) -> Iterable[U]:
+        ...
+
+    @abstractmethod
+    def seek(self, offset: int, whence: int = SEEK_SET, /) -> int:
+        pass
+
+    @abstractmethod
+    def seekable(self) -> bool:
+        ...
+
+    @abstractmethod
+    def tell(self) -> int:
+        ...
+
+    @abstractmethod
+    def truncate(self, size: Optional[int] = None) -> int:
+        ...
+
+    @abstractmethod
+    def writable(self) -> bool:
+        ...
+
+    @abstractmethod
+    def writelines(self, lines: Iterable[U], /) -> None:
+        ...
+
+    @abstractmethod
+    def read(self, size: int = -1) -> U:
+        ...
+
+    @abstractmethod
+    def write(self, b: U, /) -> int:
+        ...
+
+    def __enter__(self: S) -> S:
+        ...
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_val: Optional[BaseException],
+        exc_tb: Optional[TracebackType],
+        /,
+    ) -> Optional[bool]:
+        ...
+
+
+class BaseResourceHandle(ABC, ResourceHandleProtocol[U]):
+    """Base class interface for the handle like interface of `ResourcePath`
+    subclasses.
+
+    Parameters
+    ----------
+    mode : `str`
+        Handle modes as described in the python `io` module
+    log : `~logging.Logger`
+        Logger to used when writing messages
+    newline : `str`
+        When doing multiline operations, break the stream on given character
+        Defaults to newline.
+
+    Notes
+    -----
+    Documentation on the methods of this class line should refer to the
+    corresponding methods in the `io` module.
+    """
+
+    _closed: CloseStatus
+    _mode: str
+    _log: Logger
+    _newline: U
+
+    def __init__(self, mode: str, log: Logger, *, newline: Optional[AnyStr] = None) -> None:
+        if newline is None:
```

Comment:
We don't have any test where `newline` is `None`?

Reply:
I dont actually think we have anything that makes use of it right now, most of the child constructors want something defined